account_early_payment_discount/tests/test_payment_term.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2016 Cyril Gaudin (Camptocamp)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from psycopg2 import IntegrityError
+
+from odoo.tests.common import TransactionCase
+from odoo.tools.misc import mute_logger
+
+
+class TestPaymentTerm(TransactionCase):
+
+    @mute_logger('odoo.sql_db')
+    def test_constraint__failed(self):
+        # Logs are muted to prevent the expected integrity error from being
+        # reported in the logs, which would cause a failed test status.
+        with self.assertRaises(IntegrityError):
+            self.env['account.payment.term'].create({
+                'name': 'Unittest payment term',
+                'early_payment_discount': True,
+            })
+
+    def test_constraint__good(self):
+        self.env['account.payment.term'].create({
+            'name': 'Unittest payment term',
+            'early_payment_discount': True,
+            'epd_nb_days': 5,
+            'epd_discount': 2
+        })
```

Comment:
asserting something would be nice.

Reply:
This test simply checks that it is possible to create a payment term if the mandatory attributes 'epd_nb_days' and 'epd_discount' are provided. Which is not the case in the previous test where those attributes are omitted, causing an integrity error.
The assert is then implicit here.

Reply:
IMO you should then assert the field on the created record has the value you provided. If someone breaks the creation doing any change the test will still pass. 
Moreover "Explicit is better than implicit" PEP 20 :wink: